### PR TITLE
Support any container with begin, end, and size. Fixes #4

### LINKED ISSUE #4: Support for all kinds of containers
Use a little bit of `type_traits` magic to make debugging of all container types possible.

For example, use `std::enable_if` to use the container-specialized overload if a `begin`, `end` (and `size`) function exists.

### PR DESCRIPTION
**Pull request by hmenke**
Fixes #4.

### CODE REVIEW

File: dbg.h [hunk 1]
```diff
@@ -35,25 +35,81 @@ License (MIT):
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <type_traits>
 #include <vector>
```
Comment (sharkdp) [inline dbg.h]: We should be able to remove the <vector> include.

Comment (hmenke) [reply dbg.h]: `std::enable_if`?

**Comment by sharkdp** [comment]
Thank you very much!

File: dbg.h (hunk 1 above)
Comment (sharkdp) [reply dbg.h]: Not `type_traits`, the `vector` include.

**Pull request merged**